api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -1056,52 +1056,43 @@ static String notebookLockingEmailHash(String bucket, String email) {
     WorkspaceResourceResponse workspaceResourceResponse = new WorkspaceResourceResponse();
     workspaceResourceResponse.addAll(
         cohorts.stream()
-            .map(cohortMapper::dbModelToClient)
             .map(
                 cohort ->
-                    workspaceMapper
-                        .workspaceResourceFromDbWorkspace(workspace, workspaceAccessLevel)
-                        .cohort(cohort)
-                        .modifiedTime(cohort.getLastModifiedTime().toString()))
+                    workspaceMapper.workspaceResourceFromDbWorkspaceAndCohort(
+                        workspace, workspaceAccessLevel, cohortMapper.dbModelToClient(cohort)))
             .collect(Collectors.toList()));
     List<DbCohortReview> reviews =
         cohortReviewService.getRequiredWithCohortReviews(workspaceNamespace, workspaceId);
     workspaceResourceResponse.addAll(
         reviews.stream()
-            .map(cohortReviewMapper::dbModelToClient)
             .map(
                 cohortReview ->
-                    workspaceMapper
-                        .workspaceResourceFromDbWorkspace(workspace, workspaceAccessLevel)
-                        .modifiedTime(cohortReview.getLastModifiedTime().toString())
-                        .cohortReview(cohortReview))
+                    workspaceMapper.workspaceResourceFromDbWorkspaceAndCohortReview(
+                        workspace,
+                        workspaceAccessLevel,
+                        cohortReviewMapper.dbModelToClient(cohortReview)))
```

Comment:
@jaycarlton is there a way to do this in the mapper? Ideally it gets passed a dbModel object, and knows how to do the mapping?